src/font.rs

```diff
@@ -353,6 +353,70 @@ impl Font {
             .guidelines
             .get_or_insert_with(Default::default)
     }
+
+    /// Returns a vector of immutable lib key string references
+    pub fn lib_keys(&self) -> Vec<&String> {
+        self.lib.keys().collect::<Vec<&String>>()
+    }
+
+    /// Returns an iterator over immutable lib key string references
+    pub fn iter_lib_keys(&self) -> impl Iterator<Item = &String> {
+        self.lib.keys()
+    }
+
+    /// Returns a vector of immutable group name string references
+    pub fn group_names(&self) -> Vec<&String> {
+        match &self.groups {
+            Some(g) => g.keys().collect::<Vec<&String>>(),
+            None => vec![],
+        }
+    }
+
+    /// Returns an iterator over immutable group name string references
+    pub fn iter_group_names(&self) -> impl Iterator<Item = &String> {
+        match &self.groups {
+            Some(g) => {
+                let mut groupvec = vec![];
+                for groupname in g.keys() {
+                    groupvec.push(groupname)
+                }
+                groupvec.into_iter()
+            }
+            None => vec![].into_iter(),
+        }
+    }
+
+    /// Returns a vector of immutable kerning glyph name string references
+    pub fn kerning_names(&self) -> Vec<&String> {
+        match &self.kerning {
+            Some(k) => k.keys().collect::<Vec<&String>>(),
+            None => vec![],
+        }
+    }
+
+    /// Returns an iterator over immutable kerning glyph name string references
+    pub fn iter_kerning_names(&self) -> impl Iterator<Item = &String> {
+        match &self.kerning {
+            Some(k) => {
+                let mut kerningvec = vec![];
+                for kerningname in k.keys() {
+                    kerningvec.push(kerningname)
+                }
+                kerningvec.into_iter()
+            }
+            None => vec![].into_iter(),
+        }
+    }
+
+    /// Returns a vector of layer name String types
+    pub fn layer_names(&self) -> Vec<String> {
+        self.layers.names().map(|s| s.to_string()).collect::<Vec<String>>()
+    }
+
+    /// Returns an iterator over layer name String types
+    pub fn iter_layer_names(&self) -> impl Iterator<Item = String> + '_ {
+        self.layers.names().map(|s| s.to_string())
+    }
```

Comment:
here we have options, but if we wanted symmetry with the other API additions my suggestion would be to return `&str`, and do `self.layers.names().map(|s| s.as_ref())`.